tests/command.dev.test.js

```diff
@@ -1633,6 +1634,67 @@ export const handler = async function () {
         })
       })
     })
+
+    test(testName(`should use custom functions timeouts`, args), async (t) => {
+      await withSiteBuilder('site-with-custom-functions-timeout', async (builder) => {
+        await builder
+          .withNetlifyToml({
+            config: {
+              build: { publish: 'public' },
+              functions: { directory: 'functions' },
+            },
+          })
+          .withFunction({
+            path: 'hello.js',
+            handler: async () => {
+              await new Promise((resolve) => {
+                const SLEEP_TIME = 2000
+                setTimeout(resolve, SLEEP_TIME)
+              })
+              return {
+                statusCode: 200,
+                body: 'Hello World',
+              }
+            },
+          })
+          .buildAsync()
+
+        const siteInfo = {
+          account_slug: 'test-account',
+          id: 'site_id',
+          name: 'site-name',
+          functions_config: { timeout: 1 },
+        }
+
+        const routes = [
+          { path: 'sites/site_id', response: siteInfo },
+
+          { path: 'sites/site_id/service-instances', response: [] },
+          {
+            path: 'accounts',
+            response: [{ slug: siteInfo.account_slug }],
+          },
+        ]
+
+        await withMockApi(routes, async ({ apiUrl }) => {
+          await withDevServer(
+            {
+              cwd: builder.directory,
+              args,
+              env: {
+                NETLIFY_API_URL: apiUrl,
+                NETLIFY_SITE_ID: 'site_id',
+                NETLIFY_AUTH_TOKEN: '',
```

Comment:
We need to set `NETLIFY_AUTH_TOKEN: ''` since our CI sets it for all tests.
Having an `NETLIFY_AUTH_TOKEN` makes `@netlify/config` thinks we want to access a "real" API:
https://github.com/netlify/build/blob/b442daf2efa2a31057e8628309903f6ec3976945/packages/config/src/options/main.js#L40

Reply:
A better way to do it is https://github.com/netlify/cli/pull/2239/files#diff-e175b551bfa3b766a5f2559e0c3e037415c7034a04776cf76162ccbdcd0370aaR34